src/components/common/NetworkSelector.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <div class="network-selector">
+    <div
+      v-for="network in mainnets"
+      :key="network.chain_id"
+      class="network-item"
+      :class="{ selected: networkId === network.id }"
+      @click="network.chain_id ? selectNetworkHandler(network) : false"
+    >
+      <img
+        v-tooltip.right="{ content: network.title, offset: 8 }"
+        :src="`${network.icon}`"
+        :alt="`logo for network ${network.title}`"
+      />
+    </div>
+    <div class="network-item">
+      <i
+        v-tooltip.right="{ content: `All Networks`, offset: 8 }"
+        class="all-networks material-icons notranslate"
+        @click="
+          $route.name !== `networks` ? $router.push({ name: `networks` }) : null
+        "
+        >add</i
+      >
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapState } from "vuex"
+
+export default {
+  name: `network-selector`,
+  data: () => ({}),
+  computed: {
+    ...mapState([`connection`]),
+    ...mapGetters([`networks`]),
+    ...mapGetters({ networkId: `network` }),
+    mainnets() {
+      return this.networks.filter((network) => network.testnet === false)
+    },
+    networkSlug() {
+      return this.connection.networkSlug
+    },
+  },
+  methods: {
+    async selectNetworkHandler(network) {
```

Comment:
why not make them all router links as our router handles this logic already

Reply:
do you mean we can include which network we want to select in router params?

Reply:
just make all the network icons a router link to /network/portfolio. that would remove a lot of logic from this component

Reply:
does our router accept networks as route params?

`this.$store.dispatch(`setNetwork`, network)`